src/thunderbots/software/grsim_communication/motion_controller.cpp

```diff
@@ -18,9 +18,10 @@
 #include "util/logger/init.h"
 
 MotionController::Velocity MotionController::bangBangVelocityController(
-    const Robot robot, const Point dest, const double desired_final_speed,
-    const Angle desired_final_orientation, const double delta_time,
-    const double max_speed_meters_per_second,
+    const Robot robot,
```

Comment:
Perhaps we can create an structure to contain all the the parameters to this function other than the motion_command (something like VelocityControllerData). In my opinion this would clean up this function and make it easierto capture them all in the lambda function used below.

Thoughts @MathewMacDougall @garethellis0

Reply:
I definitely agree. @garethellis0 and I have previously considered making the MotionController a class so we can just construct it once with the 
```
const double max_speed_meters_per_second
const double max_angular_speed_radians_per_second
const double max_acceleration_meters_per_second_squared
const double max_angular_acceleration_meters_per_second_squared```
values as well since these are basically constants.

Would we still want to use a struct for the remaining parameters?

Reply:
Is there any reason why MotionController wasn't made a class? If it was a class I don't think there would be a need to create a struct since having only 3 parameters to BangBangVelocityController seems reasonable to me.

Reply:
Yep, this is a great question. We definitely considered it, but initially it took few enough parameters that it didn't matter. I think that refactoring it to a class is the best option here.

Reply:
Another reason it wasn't initially a class was because of how it was used. It was easy to just call a function and get a result, rather than having to store an instance of the MotionController. But yea I agree now that a class would probably be best here.

Reply:
ok I'll refactor into a class then

Reply:
Probably the best call, thanks for handling that. Sorry we didn't suggest it earlier :grimacing: